R/utils.R

```diff
@@ -230,3 +230,68 @@ padding_lst <- function(ft, indents) {
     flextable::padding(ft, s, 1, padding.left = (indents[s] + 1) * 10)
   }, seq_len(length(indents)), ft)
 }
+
+#' Split a text block into smaller blocks with a specified number of lines.
+#'
+#' This function takes a block of text and divides it into smaller blocks, each containing
+#' a specified number of lines.
+#'
+#' @param block_text A character vector containing the input block of text.
+#' @param n The number of lines per block.
+#'
+#' @return A list of character vectors, where each element is a smaller block of text
+#'         containing 'n' lines. If the input block of text has fewer lines than 'n', the
+#'         entire block is returned as a single element list.
+#'
+#' @keywords internal
+split_text_into_blocks <- function(block_text, n) {
+  lines <- strsplit(block_text, "\n")[[1]]
```

Comment:
Add `checkmate::assert_*` here? (if it makes sense)

Reply:
this is already been checked here https://github.com/insightsengineering/teal.reporter/blob/4fa942ec6c8c72f466505b969244c7a70d7e9f28/R/Renderer.R#L32 

I feel it's unnecessary to double-check it here.

Reply:
How? `"TextBlock"` is not a character string.

Reply:
The `split_text_into_blocks() `function is specifically employed within rcodeBlock2md to ensure that it correctly handles Rcode blocks. I've observed that the get_content() function of RcodeBlock, inherited from the contentblock R6 class, returns characters only.

However, I believe including testing in each function is a good practice. thnaks @chlebowa; I have already integrated your suggestion, incorporating checkmate::assert_string(x) based on below comment